pkg/storage/minio/saver.go

```diff
@@ -4,31 +4,54 @@ import (
 	"bytes"
 	"context"
 	"io"
+	"sync"
 
 	"github.com/gomods/athens/pkg/errors"
 	"github.com/gomods/athens/pkg/observ"
 	minio "github.com/minio/minio-go"
+	"golang.org/x/sync/errgroup"
 )
 
+type modMeta struct {
+	file string
+	len  int64
+	data io.Reader
+}
+
+// Save takes in a module and its data and tries to save it in the minio bucket
 func (s *storageImpl) Save(ctx context.Context, module, vsn string, mod []byte, zip io.Reader, info []byte) error {
 	const op errors.Op = "storage.minio.Save"
 	ctx, span := observ.StartSpan(ctx, op.String())
 	defer span.End()
+
+	var eg errgroup.Group
 	dir := s.versionLocation(module, vsn)
-	modFileName := dir + "/" + "go.mod"
-	zipFileName := dir + "/" + "source.zip"
-	infoFileName := dir + "/" + vsn + ".info"
-	_, err := s.minioClient.PutObject(s.bucketName, modFileName, bytes.NewReader(mod), int64(len(mod)), minio.PutObjectOptions{})
-	if err != nil {
-		return errors.E(op, err)
+
+	mS := [3]modMeta{
+		{file: dir + "/" + "go.mod", len: int64(len(mod)), data: bytes.NewReader(mod)},
+		{file: dir + "/" + "source.zip", len: -1, data: zip},
+		{file: dir + "/" + vsn + ".info", len: int64(len(info)), data: bytes.NewBuffer(info)},
 	}
-	_, err = s.minioClient.PutObject(s.bucketName, zipFileName, zip, -1, minio.PutObjectOptions{})
-	if err != nil {
-		return errors.E(op, err)
+
+	for _, m := range mS {
+		m := m
+		eg.Go(func() error {
+			_, err := s.minioClient.PutObject(s.bucketName, m.file, m.data, m.len, minio.PutObjectOptions{})
+			return err
+		})
 	}
-	_, err = s.minioClient.PutObject(s.bucketName, infoFileName, bytes.NewReader(info), int64(len(info)), minio.PutObjectOptions{})
-	if err != nil {
-		return errors.E(op, err)
+
+	if err := eg.Wait(); err != nil {
```

Comment:
it seems that you are nowhere returning this error to the caller, therefore we'd be telling the client the save was successful when in fact it wasn't :)

Reply:
Thanks for catching that. Fixed that!